README.md

```diff
@@ -1,3 +1,161 @@
-# Run Parasoft SOAtest action
+# Run Parasoft SOAtest
 
-This action enables you to run functional tests with Parasoft SOAtest and review results on GitHub.
\ No newline at end of file
+[![Build](https://github.com/parasoft/run-soatest-action/actions/workflows/build.yml/badge.svg)](https://github.com/parasoft/run-soatest-action/actions/workflows/build.yml)
+
+This action enables you to collect functional tests results with Parasoft SOAtest and review results directly on GitHub.
+
+Parasoft SOAtest is a testing tool that automates thorough testing for composite applications with robust support for REST and web services, including over 120 protocols/message types. It is an enterprise-grade solution that simplifies complex testing for business-critical transactions through APIs, message brokers, databases, ERPs, browser-based UIs, and other endpoints.
+- Request [a demo](https://www.parasoft.com/products/parasoft-soatest/soatest-request-a-demo/) to see an overview of Parasoft SOAtest's features and benefits.
+- See the [user guide](https://docs.parasoft.com/display/SOA20232) for information about Parasoft SOAtest's capabilities and usage.
+
+Please visit the [official Parasoft website](http://www.parasoft.com) for more information about Parasoft SOAtest and other Parasoft products.
+
+## Quick start
+
+To run your Parasoft SOAtest and review test results on GitHub, you need to customize your GitHub workflow to include:
+- The action to run SOAtest.
+- The action to upload the transformed xUnit report to GitHub.
+
+### Prerequisites
+This action requires Parasoft SOAtest with a valid Parasoft license.
+
+We recommend that you run Parasoft SOAtest on a self-hosted rather than GitHub-hosted runner.
+
+### Adding the Run SOAtest Action to a GitHub Workflow
+Add the `Run SOAtest` action to your workflow to launch test suites with Parasoft SOAtest.
+
+At a minimum, the action requires the `soatestWorkspace` parameter to be configured to specify the path to a SOAtest workspace that determines the scope of analysis.
+
+### Uploading Analysis Results to GitHub and reviewing the results
+By default, the `Run SOAtest` action generates some reports in report directory. You can upload them by adding the upload-artifact action to your workflow.
```

Comment:
I think we can state in the first paragraph that `convertReportToXUnit` is set to true as default and this action generates XML report and convert to xUnit report.
In the second paragraph, we can say that user can either upload the XML report as artifact or publish the results with recommended action which reads the converted xUnit report.